src/newgrf_text.cpp

```diff
@@ -252,7 +252,7 @@ std::string TranslateTTDPatchCodes(uint32_t grfid, uint8_t language_id, bool all
 	}
 
 	/* Helper variable for a possible (string) mapping. */
-	UnmappedChoiceList *mapping = nullptr;
+	std::optional<UnmappedChoiceList> mapping;
```

Comment:
Or maybe not... `mapping = std::nullopt;`?

Reply:
Why? We do not write `std::string s = "";` either.